go/vt/vtgate/engine/route.go

```diff
@@ -333,7 +333,8 @@ func (route *Route) execute(vcursor VCursor, bindVars, joinVars map[string]*quer
 	}
 
 	shardQueries := route.getShardQueries(route.Query, params)
-	result, err := vcursor.ExecuteMultiShard(params.ks, shardQueries, isDML)
+	// isFinal should be true only if it's a DML.
```

Comment:
This isn't necessarily obvious -- I _think_ the point is that the only way to get here is for a single-shard (or unsharded) query and therefore isFinal is true in those cases.

Could we just set it to be unconditionally true here and then handle the case where isDML is false but isFinal is true?

Reply:
I think the `isFinal` name is misleading. It should actually not be true if we're performing reads. This flag eventually reaches scatter-conn, and if it's set, it goes through the instant-commit code path, which is wasteful for reads.

I'll rename it to `canCommit`, which implies that it's relevant only for DMLs.